src/shared_key.rs

```diff
@@ -22,9 +23,85 @@ pub enum SharedKeyError {
     Generate,
 }
 
+const DITTO_IDENTITY_TAG: &str = "DITTO IDENTITY";
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct X509Auth {
+    certificates: Vec<Vec<u8>>,
+    private_key: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+enum JwtAuth {
+    AuthUrl { url: String },
+    AuthProvider { name: String },
+    Token { token: String },
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+struct ManualIdentity {
+    version: u16,
+    private_key: Vec<u8>,
+    expiry: chrono::DateTime<chrono::Utc>,
+    identity_data: Vec<u8>,
+    inband_ca_pubkey_keys: Vec<Vec<u8>>,
+    // inband_certificate: InbandCertificate,
+    // inband_metadata: Option<??>,
```

Comment:
My suggestion would be an `Option<InBandAuth>` the same as the others. We'll have a few things to group together.